### PR TITLE
Animation

### PR DESCRIPTION
**Pull request by EngOmarElsayed**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swiftui/concepts/animation/animation.md
Comment (CBID2) [inline content/swiftui/concepts/animation/animation.md:3]: [Proofreading] Developers might wonder if this describes animations. Consider using the example below to help you revise it: 
```suggestion
Description: 'In Swift, animations are a response to state changes and produce seamless visual updates.'
```

File: content/swiftui/concepts/animation/animation.md
Comment (CBID2) [inline content/swiftui/concepts/animation/animation.md:16]: [Proofreading] While mentioning the two types of animation is helpful, it would be best to just introduce them in the other sections and mostly focus on defining animations in this section.   
```suggestion
**Animation** is the ability to add smooth view changes to your view by changing the state of the view.  Adding animations to the app makes it more interactive, enhancing user experience.
```

File: content/swiftui/concepts/animation/animation.md
Comment (CBID2) [inline content/swiftui/concepts/animation/animation.md:46]: [Proofreading] Mentioning the context in which `animation` is used would make the explanation clearer. 
```suggestion
> **Note:** `.animation` in this code snippet, is a view modifier, which means that views That have this view applied to will be the only views to be animated.
```

File: content/swiftui/concepts/animation/animation.md
Comment (CBID2) [inline content/swiftui/concepts/animation/animation.md:52]: [Proofreading] Some parts of the sentence needed a bit of clarity, so consider using the example below to help you. 
```suggestion
In this simple animation, the background and the text of the button change depending on the value of the `isanimation` variable. When we press the button, the `isanimation` changes which activate the implicit animation and smoothly change the button's text and background (try to remove the animation to see the difference). As the name implies, SwiftUI creates and manages implicit animations whenever specific situations or user interface state changes take place, which means developers don't have full control over them.
```

File: content/swiftui/concepts/animation/animation.md
Comment (CBID2) [inline content/swiftui/concepts/animation/animation.md:56]: [fix]: I found some typos that need fixing.
```suggestion
To be able to make Explicit animation, we use the same three steps we used in the implicit animation. The below code snippet uses Explicit animation to change the position of a button:
```

File: content/swiftui/concepts/animation/animation.md
Comment (CBID2) [inline content/swiftui/concepts/animation/animation.md:83]:
```suggestion
As you see from the output, we used the `withAnimation` to change the poisition of the button with a spring effect to it. what is beautiful about explicit animation is that you can animate different views with the same call of `withAinmation`, but the two different views must use the same state variable to do that. For example, to animate the background color of the label view of the button and the position of the button at the same time, a ternary operator will be added to `.background()` modifier. It looks like this:
```

File: content/swiftui/concepts/animation/animation.md
Comment (CBID2) [inline content/swiftui/concepts/animation/animation.md:107]:
```suggestion
One thing to consider is that no limit on how `withAnimation` or `.animation` can be used. This is a very powerful tool that can be used in many ways. 
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @EngOmarElsayed. I gave some suggestions on how you can improve your entry. In the future, I highly recommend using a tool like Grammarly to help you revise your future entries for misspellings and other grammatical errors. Other than that, you're off to a great start! :) (Note: I'm not a maintainer. Just a friendly contributor who does code reviews)

File: content/swiftui/concepts/animation/animation.md
Comment (EngOmarElsayed) [reply content/swiftui/concepts/animation/animation.md:46]: this is a general rule not only about this code snip

**Review by yangc95 — CHANGES_REQUESTED** [review]
@EngOmarElsayed main feedback is to remove the safe areas from the gifs provided. Otherwise, I'll do one last check for the code blocks and will approve.

😄 thanks for being thorough with the examples

File: content/swiftui/concepts/animation/animation.md
Comment (CBID2) [inline content/swiftui/concepts/animation/animation.md:52]: [Proofreading] The first half of the sentence needs clarity. Consider using the suggestion I placed below: 
```suggestion
In this simple animation, the background and the text of the button change depending on the value of the `isanimation` variable. When the button is pressed, `isanimation` changes, causing the implicit animation to change the text and background of the button. As the name implies, SwiftUI creates and manages implicit animations whenever specific situations or user interface state changes take place, which means developers don't have full control over the animations. With explicit animation, developers have full control over them.
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @EngOmarElsayed. I found something that needs revision. Keep going! :)

**Comment by EngOmarElsayed** [comment]
> @EngOmarElsayed main feedback is to remove the safe areas from the gifs provided. Otherwise, I'll do one last check for the code blocks and will approve.
> 
> 😄 thanks for being thorough with the examples

Can you please tell me what part you are talking about ? 😅

**Comment by yangc95** [comment]
@EngOmarElsayed I'm marking this as ready for a second review cc: @KTom101 

Just FYI, I will be changing out the media for ones that crop out the safe areas.

**Review by KTom101 — CHANGES_REQUESTED** [review]
@EngOmarElsayed I've added some corrections to this entry.

File: content/swiftui/concepts/animation/animation.md
Comment (EngOmarElsayed) [reply content/swiftui/concepts/animation/animation.md:59]: Sorry for that my keyboard is broken

**Review by KTom101 — APPROVED** [review]
@EngOmarElsayed I added a couple more changes directly and ran `npm run format:all` to fix the failed [checks](https://www.codecademy.com/resources/docs/contribution-guide/getting-set-up-on-your-computer#:~:text=There%20are%20continuous%20integration%20(CI)%20workflows). Looks good to merge, thank you.

**Review by yangc95 — APPROVED** [review]

**Pull request merged**
